pkg/kubelet/kubelet.go

```diff
@@ -1817,7 +1823,7 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 	}
 
 	// Fetch the pull secrets for the pod
-	pullSecrets, err := kl.getPullSecretsForPod(pod)
```

Comment:
@liggitt , I have known your meaning of short term, you said that we would only make a local cache inner the func getPullSecretsForPod(pod) while syncing the single pod, this change would prevent duplicate secrets being retrieved to a certain extent;